### PR TITLE
[dag] support for Quorum Store

### PR DESCRIPTION
**Pull request by ibalajiarun**

### CODE REVIEW

**Comment by ibalajiarun** [comment]
Current dependencies on/for this PR:
* main
  * **PR #10890** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10890?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #10918** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10918?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #10799** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10799?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10918?utm_source=stack-comment).

File: consensus/src/dag/dag_store.rs [hunk 1]
```diff
@@ -161,6 +174,8 @@ impl Dag {
         self.storage.save_certified_node(&node)?;
         debug!("Added node {}", node.id());
         round_ref[index] = Some(NodeStatus::Unordered(node.clone()));
+        self.payload_manager
+            .prefetch_payload_data(node.payload(), node.metadata().timestamp());
```
Comment (sasha8) [inline consensus/src/dag/dag_store.rs:178]: The crash recover and fetch paths go from here as well, right?

File: consensus/src/payload_manager.rs [hunk 2]
```diff
@@ -22,13 +22,23 @@ use futures::{channel::mpsc::Sender, SinkExt};
 use std::sync::Arc;
 use tokio::sync::oneshot;
 
+pub trait TPayloadManager: Send + Sync {
+    fn prefetch_payload_data(&self, payload: &Payload, timestamp: u64); 
+}
+
```
Comment (sasha8) [inline consensus/src/payload_manager.rs:28]: Why do we need the trait?
Testing?

**Review by sasha8 — APPROVED** [review]

File: consensus/src/dag/dag_store.rs (hunk 1 above)
Comment (zekun000) [reply consensus/src/dag/dag_store.rs:178]: it does, also the prefetch needs to tolerate multiple calls as well.

**Review by zekun000 — APPROVED** [review]

File: consensus/src/payload_manager.rs (hunk 2 above)
Comment (ibalajiarun) [reply consensus/src/payload_manager.rs:28]: yes, testing

**Pull request merged**